tracer/src/Datadog.Trace.Tools.Runner/AnalyzeInstrumentationErrorsCommand.cs

```diff
@@ -0,0 +1,143 @@
+            return -1;
+        }
+
+        var generatorArgs = new AssemblyGeneratorArgs(processLogDir, modulesToVerify: null);
+
+        var exportedModulesPathsAndMethods = InstrumentedAssemblyGeneration.Generate(generatorArgs);
+
+        bool allVerificationsPassed = true;
+
+        foreach (var instrumentedAssembly in exportedModulesPathsAndMethods)
+        {
+            var result = new VerificationsRunner(instrumentedAssembly.InstrumentedAssemblyPath, instrumentedAssembly.OriginalAssemblyPath, instrumentedAssembly.InstrumentedMethods.Select(m => (m.MethodName, m.TypeName)).ToList()).Run();
+            AnsiConsole.WriteLine($"Verification for {instrumentedAssembly.InstrumentedAssemblyPath} {(result.IsValid ? "passed." : "failed: " + result.FailureReason)}");
+            allVerificationsPassed = allVerificationsPassed && result.IsValid;
+        }
+
+        try
+        {
+            foreach (var instrumented in exportedModulesPathsAndMethods)
+            {
+                foreach (var method in instrumented.InstrumentedMethods)
+                {
+                    AnsiConsole.WriteLine("--------------------------------------------------------");
+                    AnsiConsole.WriteLine($"Method {method.FullName} has been instrumented.");
+                    AnsiConsole.WriteLine($"Instrumented IL Code:{Environment.NewLine}{method.Instructions}");
+                    AnsiConsole.WriteLine($"Instrumented Decompiled Code:{Environment.NewLine}{method.DecompiledCode.Value}");
+                }
+            }
+
+            var checkProcessReport = AnsiConsole.ExportText();
+            await Task.Yield();
+        }
+        catch (Exception e)
+        {
+            AnsiConsole.WriteLine($"Error while uploading the report: {e}");
+            return -1;
+        }
+
+        return allVerificationsPassed ? 0 : -1;
+    }
+
+    private string GetProcessLogDirectory(string baseLogDir, AnalyzeInstrumentationErrorsSettings settings)
+    {
+        var dirs = Directory.EnumerateDirectories(baseLogDir).Select(d => new DirectoryInfo(d)).ToList();
+        if (dirs.Count == 0)
+        {
+            return null;
+        }
+
+        if (dirs.Count == 1)
+        {
+            if (string.IsNullOrEmpty(settings.ProcessName) && settings.Pid == null)
+            {
+                return dirs[0].FullName;
+            }
+        }
+
+        if (string.IsNullOrEmpty(settings.ProcessName) && settings.Pid == null)
+        {
+            return null;
+        }
+
+        var processName = string.IsNullOrEmpty(settings.ProcessName) ? "[A-Z0-9]" : $"({settings.ProcessName})";
+        var pid = settings.Pid == null ? "\\d+" : settings.Pid.ToString();
+        var pattern = $"^{processName}(_){pid}(_)[0-9-_]+$";
+        return dirs.SingleOrDefault(d => Regex.IsMatch(d.Name, pattern))?.FullName;
+    }
+
+    private string GetLogDirectory(string logDirectory, int? pid)
+    {
+        if (!string.IsNullOrEmpty(logDirectory))
```

Comment:
I'd recommend moving this if statement to the call site [as I proposed here](https://github.com/DataDog/dd-trace-dotnet/pull/3408/files#r1014828627) and removing the parameter `logDirectory`) as it would probably make the purpose of this method clearer to the reader.

Reply:
Fixed in d6e34e581c70cc2271806c12db2c3d4f2caf78f1